packages/dev/s2-docs/src/NavigationSuspense.tsx

```diff
@@ -0,0 +1,70 @@
+'use client';
+
+import type {Page} from '@parcel/rsc';
+import {PageSkeleton} from './PageSkeleton';
+import React, {Suspense} from 'react';
+
+let navigationPromise: Promise<void> | null = null;
+let navigationResolve: (() => void) | null = null;
+let targetPathname: string | null = null;
+
+export function setNavigationLoading(loading: boolean, pathname?: string) {
+  if (loading) {
+    targetPathname = pathname || null;
+    navigationPromise = new Promise(resolve => {
+      navigationResolve = resolve;
```

Comment:
Is there like some kind of minimum time that the skeleton appears for? I noticed that when switching between pages the table of contents appears instantly but the content doesn't - it always shows the skeleton for a bit even if the page is already cached.